lib/cancan/rule.rb

```diff
@@ -34,11 +34,13 @@ def initialize(base_behavior, action, subject, *extra_args, &block)
     def inspect
       repr = "#<#{self.class.name}"
       repr += "#{@base_behavior ? 'can' : 'cannot'} #{@actions.inspect}, #{@subjects.inspect}, #{@attributes.inspect}"
-      repr += if with_scope?
-                ", #{@conditions.where_values_hash}"
-              elsif [Hash, String].include?(@conditions.class)
-                ", #{@conditions.inspect}"
-              end
+
+      if with_scope?
```

Comment:
@frostblooded  I guess you have pushed this file by mistake.

Reply:
No. This is the actual fix of the bug. The way it was previously written, when there was an SQL condition, the if/elsif wouldn't enter either of the tho blocks and it would "return" nil, which would make the code attempt to do a "repr += nil" and would throw and error.